#VRML_SIM R2023b utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/floors/protos/RectangleArena.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/appearances/protos/BrushedAluminium.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/appearances/protos/BakelitePlastic.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/appearances/protos/BrushedSteel.proto"

WorldInfo {
}
Viewpoint {
  orientation 0.006554062501261762 -0.14018386543754968 -0.9901037966474608 6.188384658499786
  position -4.5455673219294095 -0.8649356546282125 3.4932024967623923
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  floorSize 4 4
}
Robot {
  translation -0.285783 0.359373 1.49001
  rotation 0.3315489812804135 0.14085499204718654 -0.9328639473295703 0.0132564
  children [
    DEF Base Shape {
      appearance BrushedAluminium {
      }
      geometry Box {
        size 1.2 2.5 0.05
      }
    }
    Solid {
      translation 0 1.22 1.235
      children [
        DEF Base2 Shape {
          appearance BrushedAluminium {
          }
          geometry Box {
            size 1.2 0.05 2.5
          }
        }
        HingeJoint {
          jointParameters HingeJointParameters {
            position -3.901380562438175e-05
            axis 0 1 0
            anchor 0.44 -0.06 0.6
          }
          device [
            RotationalMotor {
              name "motor1"
            }
            PositionSensor {
            }
          ]
          endPoint Solid {
            translation 0.4400049999999948 -0.06 0.6000000002268422
            rotation 0.9999999996194797 -1.95069551136704e-05 1.9506902807242348e-05 1.5707936457472857
            children [
              DEF connector1 Shape {
                appearance PBRAppearance {
                  baseColor 0 0 0
                }
                geometry Cylinder {
                  height 0.05
                  radius 0.05
                }
              }
              Solid {
                translation 0 0 1.02
                children [
                  DEF ARM Shape {
                    appearance PBRAppearance {
                      baseColor 1 1 0
                    }
                    geometry Box {
                      size 0.05 0.1 2
                    }
                  }
                  HingeJoint {
                    jointParameters HingeJointParameters {
                      position -6.205630412464699e-08
                      anchor -0.05 9.58629e-09 0.95
                    }
                    device [
                      RotationalMotor {
                        name "motor2"
                        maxTorque 20
                      }
                      PositionSensor {
                      }
                    ]
                    endPoint Solid {
                      translation -0.05 8.984074225493951e-10 0.9499999999999992
                      rotation -3.183046876657778e-08 -0.999999999999999 3.18304363117603e-08 1.5707953071795746
                      children [
                        DEF connector2 Shape {
                          appearance PBRAppearance {
                            baseColor 0 0 0
                          }
                          geometry Cylinder {
                            height 0.05
                            radius 0.05
                          }
                        }
                        Solid {
                          translation 0 -0.2 0.04
                          children [
                            DEF arm1 Shape {
                              appearance BakelitePlastic {
                              }
                              geometry Box {
                                size 0.1 0.5 0.05
                              }
                            }
                            HingeJoint {
                              jointParameters HingeJointParameters {
                                axis 0 0 1
                                anchor 0 -0.2 0.05
                              }
                              device [
                                RotationalMotor {
                                  name "motor3"
                                }
                                DEF roll PositionSensor {
                                  name "roll"
                                }
                              ]
                              endPoint DEF roll Solid {
                                translation 0 -0.2 0.05
                                children [
                                  DEF connector3 Shape {
                                    appearance PBRAppearance {
                                      baseColor 0 0 0
                                    }
                                    geometry Cylinder {
                                      height 0.05
                                      radius 0.05
                                    }
                                  }
                                  Solid {
                                    translation 0 0 0.05
                                    children [
                                      DEF arm2 Shape {
                                        appearance BakelitePlastic {
                                        }
                                        geometry Box {
                                          size 1 0.1 0.05
                                        }
                                      }
                                      Solid {
                                        translation 0.48 1.40785e-07 0.24
                                        rotation 0 0 1 4.692820414042842e-06
                                        children [
                                          DEF arm2(1) Shape {
                                            appearance BakelitePlastic {
                                            }
                                            geometry Box {
                                              size 0.05 0.1 1
                                            }
                                          }
                                          HingeJoint {
                                            jointParameters HingeJointParameters {
                                              anchor -0.06 0 0.45
                                            }
                                            device [
                                              RotationalMotor {
                                                name "motor4"
                                                maxTorque 20
                                              }
                                              DEF pitch PositionSensor {
                                                name "pitch"
                                              }
                                            ]
                                            endPoint Solid {
                                              translation -0.48 0 0.45
                                              children [
                                                DEF CAMERA Shape {
                                                  appearance BrushedSteel {
                                                  }
                                                  geometry Box {
                                                    size 0.9 0.5 0.5
                                                  }
                                                }
                                                InertialUnit {
                                                  name "inertial_unit"
                                                  boundingObject USE CAMERA
                                                  physics Physics {
                                                  }
                                                }
                                                Gyro {
                                                  boundingObject USE CAMERA
                                                  physics Physics {
                                                  }
                                                }
                                                Camera {
                                                  translation 1.93726e-07 0.15 0.24
                                                  rotation 0.5773509358560258 0.577349935856137 0.577349935856137 -2.094395307179586
                                                  name "CAM"
                                                  width 640
                                                  height 320
                                                  lens Lens {
                                                  }
                                                  focus Focus {
                                                  }
                                                  zoom Zoom {
                                                  }
                                                  recognition Recognition {
                                                  }
                                                  lensFlare LensFlare {
                                                  }
                                                }
                                              ]
                                              boundingObject USE CAMERA
                                              physics Physics {
                                              }
                                            }
                                          }
                                        ]
                                        boundingObject USE arm2(1)
                                        physics Physics {
                                        }
                                      }
                                      Solid {
                                        translation -0.48 -4.36445e-06 0.24
                                        rotation 0 0 1 -3.1415853071795863
                                        children [
                                          DEF arm2(1) Shape {
                                            appearance BakelitePlastic {
                                            }
                                            geometry Box {
                                              size 0.05 0.1 1
                                            }
                                          }
                                          HingeJoint {
                                            jointParameters HingeJointParameters {
                                              anchor -0.06 0 0.45
                                            }
                                            device [
                                              RotationalMotor {
                                                name "motor4"
                                              }
                                            ]
                                          }
                                        ]
                                        name "solid(1)"
                                        boundingObject USE arm2(1)
                                        physics Physics {
                                        }
                                      }
                                    ]
                                    boundingObject USE arm2
                                    physics Physics {
                                    }
                                  }
                                ]
                                boundingObject USE connector3
                                physics Physics {
                                }
                              }
                            }
                          ]
                          boundingObject USE arm1
                          physics Physics {
                          }
                        }
                      ]
                      boundingObject USE connector2
                      physics Physics {
                      }
                    }
                  }
                ]
                boundingObject USE ARM
                physics Physics {
                }
              }
            ]
            boundingObject USE connector1
            physics Physics {
            }
          }
        }
      ]
      boundingObject USE Base2
      physics Physics {
      }
    }
  ]
  boundingObject USE Base
  controller "remote3"
}
